ush/fv3gfs_downstream_nems.sh

```diff
@@ -1,294 +1,177 @@
+    # if final record is land, add next record icec
     # grep returns 1 if no match is found, so temporarily turn off exit on non-zero rc
     set +e
-    $WGRIB2 -d $end $tmpfile | egrep -i "ugrd|ustm|uflx|u-gwd"
-    export rc=$?
-    set_strict
-    if [[ $rc -eq 0 ]] ; then
-      export end=$(expr ${end} + 1)
-    elif [[ $rc -gt 1 ]]; then
-      echo "FATAL: WGRIB2 failed with error code ${rc}"
-      exit $rc
-    fi
-    # if final record is land, add next record icec 
-    set +e
-    $WGRIB2 -d $end $tmpfile | egrep -i "land"
-    export rc=$?
+    # shellcheck disable=SC2312
+    ${WGRIB2} -d "${last}" "${tmpfile}" | grep -E -i "ugrd|ustm|uflx|u-gwd|land"
+    rc=$?
     set_strict
-    if [[ $rc -eq 0 ]] ; then
-      export end=$(expr ${end} + 1)
-    elif [[ $rc -gt 1 ]]; then
-      echo "FATAL: WGRIB2 failed with error code ${rc}"
-      exit $rc
+    if (( rc == 0 )); then  # Matched the grep
+      last=$(( last + 1 ))
     fi
-    if [ $iproc -eq $nproc ]; then
-      export end=$ncount
+    if (( iproc == nproc )); then
+      last=${ncount}
     fi
 
-    $WGRIB2 $tmpfile -for ${start}:${end} -grib ${tmpfile}_${iproc}
+    # Break tmpfile into processor specific chunks in preparation for MPMD
+    ${WGRIB2} "${tmpfile}" -for "${first}":"${last}" -grib "${tmpfile}_${iproc}"
     export err=$?; err_chk
-    echo "${GFSDWNSH:-$USHgfs/fv3gfs_dwn_nems.sh} ${tmpfile}_${iproc} $fhr3 $iproc $nset" >> $DATA/poescript
+    input_file="${tmpfile}_${iproc}"
+    output_file_prefix="pgb2${grp}file_${fhr3}_${iproc}"
+    echo "${GFSDWNSH} ${input_file} ${output_file_prefix}" >> "${DATA}/poescript"
 
     # if at final record and have not reached the final processor then write echo's to
     # poescript for remaining processors
-    if [[ $end -eq $ncount ]] ;then
-      while [[ $iproc -lt $nproc ]];do
-        export iproc=$(expr $iproc + 1)
-        echo "/bin/echo $iproc" >> $DATA/poescript
+    if (( last == ncount )); then
+      for (( pproc = iproc+1 ; pproc < nproc ; pproc++ )); do
+        echo "/bin/echo ${pproc}" >> "${DATA}/poescript"
       done
       break
     fi
-    export iproc=$(expr $iproc + 1)
-  done
+  done  # for (( iproc = 1 ; iproc <= nproc ; iproc++ )); do
 
-  date
-  chmod 775 $DATA/poescript
-  export MP_PGMMODEL=mpmd
-  export MP_CMDFILE=$DATA/poescript
-  launcher=${APRUN_DWN:-"aprun -j 1 -n 24 -N 24 -d 1 cfp"}
-  if [ $machine = WCOSS2 ] ; then
-    $launcher $MP_CMDFILE
-  elif [ $machine = HERA -o $machine = ORION -o $machine = JET -o $machine = S4 ] ; then
-    if [ -s $DATA/poescript_srun ]; then rm -f $DATA/poescript_srun; fi
-    touch $DATA/poescript_srun
-    nm=0
-    cat $DATA/poescript | while read line; do
-      echo "$nm $line" >> $DATA/poescript_srun
-      nm=$((nm+1))
-    done
-    nm=$(wc -l < $DATA/poescript_srun)
-    ${launcher:-"srun --export=ALL"} -n $nm --multi-prog $DATA/poescript_srun
+  # Run with MPMD or serial
+  if [[ "${USE_CFP:-}" = "YES" ]]; then
```

Comment:
I chose to keep it this way for now since it makes it clearer at the job-level whether it will do CFP or not.  
We could push this down, but I am not seeing obvious benefit/clarity.  TBD.